Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -8,10 +8,19 @@
 
 import SourceKittenFramework
 
+public enum RuleKind: String {
+    case lint
+    case idiomatic
+    case style
+    case metrics
+    case performance
+}
+
 public protocol Rule {
     init() // Rules need to be able to be initialized with default values
     init(configuration: Any) throws
     static var description: RuleDescription { get }
+    static var kind: RuleKind { get }
```

Comment:
I still think that from an user point of view, it's better that a rule can only belong to one category.

This way we could easily in the future allow enabling/disabling full categories that once. That would be tricky if a rule was in two groups: what would be the expected behavior?